assets/javascripts/discourse/components/minimal-gamification-leaderboard.js

```diff
@@ -0,0 +1,65 @@
+import Component from "@ember/component";
+import discourseComputed from "discourse-common/utils/decorators";
+import { action } from "@ember/object";
+import LoadMore from "discourse/mixins/load-more";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default Component.extend(LoadMore, {
+  tagName: "",
+  eyelineSelector: ".user",
+  page: 1,
+  loading: false,
+  canLoadMore: true,
+  period: "all",
+  notTop10: true,
+
+  init() {
+    this._super(...arguments);
+
+    ajax(`/leaderboard`)
+      .then((model) => {
+        this.set("model", model);
+      })
+      .catch(() => this.replaceWith("/404"));
+  },
+
+  @discourseComputed("model.users")
+  currentUserRanking() {
+    const user = this.model?.personal;
+    if (user) {
+      this.set("notTop10", user.position > 10);
+    }
+    return user || null;
+  },
+
+  @discourseComputed("model.users.[]")
+  ranking(users) {
+    users?.forEach((user) => {
+      if (user.id === this.currentUser?.id) {
+        user.isCurrentUser = "true";
+      }
+    });
+    return users?.slice(0, 10);
```

Comment:
Another option would be to pass `limit=10` to the ajax call, then we wouldn't need to filter clientside.